src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -68,10 +70,19 @@ default URLDownload getUrlDownload(String query) throws MalformedURLException, F
         }
 
         try (InputStream stream = getUrlDownload(query).asInputStream()) {
-            List<BibEntry> fetchedEntries = getParser().parseEntries(stream);
+            List<BibEntry> fetchedEntries = new ArrayList<>();
 
-            // Post-cleanup
-            fetchedEntries.forEach(this::doPostCleanup);
+            // check if there is anything to read since mEDRA '404 not found' returns nothing
```

Comment:
This calls for discussing good and bad deisgn. - Why should a general class handle special cases of a sub clasS?

The mEDRA `getUrlDownload` should provide a proper stream. We will see it below.

First idea: It can be wrapped into a BufferedStream and checked if there is something and reset if everythign is OK. If not, an empty stream should be returend, which should lead to an empty list of bib entries, doesn't it?

Reply:
This was the first implementation i did, then I've overridden performSearchById inside Medra class.

Reply:
Then please revert this changes here, as it's no longer necessary, right?